source/crud/crud.md

```diff
@@ -768,7 +958,23 @@ filter, skip, and limit are added as options to the `aggregate` command.
 In the event this aggregation is run against an empty collection, an empty array will be returned with no `n` field.
 Drivers MUST interpret this result as a `0` count.
 
-##### Combining Limit and Batch Size for the Wire Protocol
+##### findOne API details
+
+The `findOne` operation is implemented using a `find` operation, but only returns the first document returned in the
+cursor. Due to the special case, `findOne` does not support the following options supported in `find`:
+
+- `batchSize`: drivers MUST NOT set a `batchSize` in the `find` command created for a `findOne` operation
+- `cursorType`: `findOne` only supports non-tailable cursors
+- `limit`: drivers MUST set `limit` to 1 in the `find` command created for a `findOne` operation
```

Comment:
Seems like a good place to mention `singleBatch` should be set right?

This should be in the tests too I think

Reply:
Wouldn't `singleBatch` be unnecessary if we're using `limit` 1?

Reply:
The node driver has always set that field for find one, maybe it is redundant but then again why does setting that field not close the cursor with the single batch it is requested to return? 

I would assume that `singleBatch == true` leads to `cursorId: 0` every time

Reply:
To summarise the discussion we had about this, `singleBatch: true` does indeed lead to `cursorId: 0` every time, including in those cases when not all requested results would fit into the result document due to the BSON size limit. In those cases, our expectation would be to receive a result cursor with the rest of the documents, but `singleBatch: true` would lead to the result omitting those documents.
For that reason, we decided to only require `singleBatch` for `findOne`, where we don't expect to receive a cursor in any case. The case of `limit == batchSize` is solved by increasing the `batchSize` by one to ensure the server encounters an end-of-stream from the PlanExecutor while trying to fill the requested batch. The server is considering a change to this behaviour as part of [SERVER-96643](https://jira.mongodb.org/browse/SERVER-96643).